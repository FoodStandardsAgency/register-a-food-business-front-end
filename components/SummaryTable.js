import React from "react";
import { Table } from "govuk-react";

const SummaryTable = props => (
  <Table
    body={
      <React.Fragment>
        {props.establishment_first_line ? (
          <Table.Row id="establishmentAddress">
            <Table.CellHeader>Establishment address</Table.CellHeader>
            <Table.Cell>
              <div id="establishmentFirstLine">
                {props.establishment_first_line}
              </div>
              <div id="establishmentStreet">
                {props.establishment_street || null}
              </div>
              <div id="establishmentTown">
                {props.establishment_town || null}
              </div>
              <div id="establishmentPostcode">
                {props.establishment_postcode}
              </div>
            </Table.Cell>
          </Table.Row>
        ) : null}
      </React.Fragment>
    }
  />
);

export default SummaryTable;
