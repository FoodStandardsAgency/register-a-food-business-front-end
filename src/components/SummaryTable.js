import React from "react";
import { Table, Paragraph } from "govuk-react";
import PropTypes from "prop-types";
import * as COLOUR from "govuk-colours";
import styled from "react-emotion";
import {
  FONT_SIZE,
  LINE_HEIGHT,
  MEDIA_QUERIES,
  NTA_LIGHT
} from "@govuk-react/constants";

const StyledTableRow = styled("div")({
  fontFamily: NTA_LIGHT,
  WebkitFontSmoothing: "antialiased",
  MozOsxFontSmoothing: "grayscale",
  fontWeight: 400,
  display: "inline",
  textTransform: "none",
  fontSize: FONT_SIZE.SIZE_16,
  lineHeight: LINE_HEIGHT.SIZE_16,
  [MEDIA_QUERIES.LARGESCREEN]: {
    fontSize: FONT_SIZE.SIZE_19,
    lineHeight: LINE_HEIGHT.SIZE_19
  },
  color: `${COLOUR.GREY_1}`
});

const OperatorDetailsTable = props => (
  <React.Fragment>
    <Table.Row TITLE>
      <Table.CellHeader>
        <Paragraph mb={0}>**Operator details**</Paragraph>
      </Table.CellHeader>
      <Table.Cell>
        <div />
      </Table.Cell>
    </Table.Row>

    {props.operator_company_name ? (
      <Table.Row id="operatorCompanyNameRow">
        <Table.CellHeader>Company name</Table.CellHeader>
        <Table.Cell>
          <div id="operator_company_name" className="bold">
            {props.operator_company_name}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_company_house_number ? (
      <Table.Row id="operatorCompaniesHouseRow">
        <Table.CellHeader>Company number</Table.CellHeader>
        <Table.Cell>
          <div id="operator_company_house_number" className="bold">
            {props.operator_company_house_number}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_charity_name ? (
      <Table.Row id="operatorCharityNameRow">
        <Table.CellHeader>Charity name</Table.CellHeader>
        <Table.Cell>
          <div id="operator_charity_name" className="bold">
            {props.operator_charity_name}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_charity_number ? (
      <Table.Row id="operatorCharityNumberRow">
        <Table.CellHeader>Charity number</Table.CellHeader>
        <Table.Cell>
          <div id="operator_charity_number" className="bold">
            {props.operator_charity_number}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_first_name ? (
      <Table.Row id="operatorNameRow">
        <Table.CellHeader>Name</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <span id="operator_first_name">{props.operator_first_name}</span>{" "}
            <span id="operator_last_name">{props.operator_last_name}</span>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_first_line ? (
      <Table.Row id="operatorAddressRow">
        <Table.CellHeader>Operator address</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="operator_first_line">{props.operator_first_line}</div>
            <div id="operator_street">{props.operator_street || null}</div>
            <div id="operator_town">{props.operator_town || null}</div>
            <div id="operator_postcode">{props.operator_postcode}</div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_primary_number ? (
      <Table.Row id="operatorContactDetailsRow">
        <Table.CellHeader>Contact number</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="operator_primary_number">
              {props.operator_primary_number}
            </div>
            <div id="operator_secondary_number">
              {props.operator_secondary_number || null}
            </div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_type ? (
      <Table.Row id="operatorTypeRow">
        <Table.CellHeader>Operator type</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="operator_type">
            {props.operator_type}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_email ? (
      <Table.Row id="operatorEmailRow">
        <Table.CellHeader>Email address</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="operator_email">
            {props.operator_email}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.contact_representative_email ? (
      <Table.Row id="contactRepresentativeRow">
        <Table.CellHeader>Email address</Table.CellHeader>
        <Table.Cell>
          <StyledTableRow
            className="contact_representative_name"
            id="contact_representative_name"
          >
            {props.contact_representative_name}{" "}
          </StyledTableRow>
          <StyledTableRow
            className="contact_representative_role"
            id="contact_representative_role"
          >
            {props.contact_representative_role
              ? `\u2022 ${props.contact_representative_role}`
              : null}
          </StyledTableRow>
          <div className="bold" id="contact_representative_name">
            {props.contact_representative_number}
          </div>
          <div className="bold" id="contact_representative_role">
            {props.contact_representative_email}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}
  </React.Fragment>
);

const EstablishmentDetailsTable = props => (
  <React.Fragment>
    <Table.Row TITLE>
      <Table.CellHeader>
        <Paragraph mb={0}>**Establishment details**</Paragraph>
      </Table.CellHeader>
      <Table.Cell>
        <div />
      </Table.Cell>
    </Table.Row>

    {props.establishment_trading_name ? (
      <Table.Row id="establishmentTradingNameRow">
        <Table.CellHeader>Establishment name</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="establishment_trading_name">
            {props.establishment_trading_name}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.establishment_first_line ? (
      <Table.Row id="establishmentAddressRow">
        <Table.CellHeader>Establishment address</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="establishment_first_line">
              {props.establishment_first_line}
            </div>
            <div id="establishment_street">
              {props.establishment_street || null}
            </div>
            <div id="establishment_town">
              {props.establishment_town || null}
            </div>
            <div id="establishment_postcode">
              {props.establishment_postcode}
            </div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.establishment_primary_number ? (
      <Table.Row id="establishmentContactDetailsRow">
        <Table.CellHeader>Contact number</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="establishment_primary_number">
              {props.establishment_primary_number}
            </div>
            <div id="establishment_secondary_number">
              {props.establishment_secondary_number || null}
            </div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.establishment_email ? (
      <Table.Row id="establishmentEmailRow">
        <Table.CellHeader>Email address</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="establishment_email">
            {props.establishment_email || null}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}
  </React.Fragment>
);

const FoodActivitiesTable = props => (
  <React.Fragment>
    <Table.Row TITLE>
      <Table.CellHeader>
        <Paragraph mb={0}>**Food activities**</Paragraph>
      </Table.CellHeader>
      <Table.Cell>
        <div />
      </Table.Cell>
    </Table.Row>

    {props.customer_type ? (
      <Table.Row id="activitiesCustomersRow">
        <Table.CellHeader>Customers</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="customer_type">
            {props.customer_type}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}
  </React.Fragment>
);

const SummaryTable = props => (
  <Table
    body={
      <React.Fragment>
        <OperatorDetailsTable {...props} />
        <EstablishmentDetailsTable {...props} />
        <FoodActivitiesTable {...props} />
      </React.Fragment>
    }
  />
);

export default SummaryTable;

SummaryTable.propTypes = {
  operator_company_name: PropTypes.string,
  operator_company_house_number: PropTypes.string,
  operator_charity_name: PropTypes.string,
  operator_charity_number: PropTypes.string,
  operator_first_line: PropTypes.string,
  operator_street: PropTypes.string,
  operator_town: PropTypes.string,
  operator_postcode: PropTypes.string,
  operator_first_name: PropTypes.string,
  operator_last_name: PropTypes.string,
  registration_role: PropTypes.string,
  establishment_trading_name: PropTypes.string,
  establishment_first_line: PropTypes.string,
  establishment_street: PropTypes.string,
  establishment_town: PropTypes.string,
  establishment_postcode: PropTypes.string,
  customer_type: PropTypes.string
};
