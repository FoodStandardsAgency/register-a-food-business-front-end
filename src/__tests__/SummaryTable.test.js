import SummaryTable from "../components/SummaryTable";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion));

// a complete list of the summary table row IDs

const mandatoryTableRows = [
  "establishmentAddressRow",
  "establishmentAddressTypeRow",
  "operatorAddressRow",
  "establishmentTradingNameRow",
  "operatorNameRow",
  "operatorTypeRow",
  "operatorCompanyNameRow",
  "operatorCompaniesHouseRow",
  "operatorCharityNameRow",
  "activitiesCustomersRow",
  "activitiesBusinessImportExportRow",
  "operatorContactDetailsRow",
  "operatorEmailRow",
  "establishmentContactDetailsRow",
  "establishmentEmailRow",
  "contactRepresentativeRow",
  "establishmentOpeningDateRow",
  "businessTypeRow",
  "businessOtherDetailsRow",
  "establishmentOpeningDaysRow"
];

const editableTableRows = [
  "establishmentAddressTypeRow",
  "establishmentTradingNameRow",
  "operatorTypeRow",
  "operatorNameRow",
  "operatorCompanyNameRow",
  "operatorCompaniesHouseRow",
  "operatorCharityNameRow",
  "operatorCharityNumberRow",
  "operatorContactDetailsRow",
  "operatorEmailRow",
  "establishmentContactDetailsRow",
  "establishmentEmailRow",
  "contactRepresentativeRow",
  "activitiesBusinessImportExportRow",
  "businessTypeRow",
  "activitiesCustomersRow",
  "businessOtherDetailsRow",
  "establishmentOpeningDaysRow"
];

// (only optional if it's optional within that page. Does not apply to pages that are optional or could be skipped.)
const optionalTableRows = ["operatorCharityNumberRow"];

const allTableRows = mandatoryTableRows.concat(optionalTableRows);

// the complete set of possible mandatory answer fields with example data
const testMandatoryAnswers = {
  operator_type: "Sole trader",
  establishment_first_line: "Example first line",
  establishment_postcode: "AA11 1AA",
  operator_first_line: "Example first line",
  operator_postcode: "AA11 1AA",
  establishment_trading_name: "Example trading name",
  operator_first_name: "John",
  operator_last_name: "Appleseed",
  operator_company_name: "Company name",
  operator_company_house_number: "AA123456",
  operator_charity_name: "Charity name",
  customer_type: "End consumer and Other buisnesses",
  import_export_activities: "None",
  operator_primary_number: "1234567",
  operator_email: "operator@email.com",
  establishment_primary_number: "12345678",
  establishment_email: "establishment@email.com",
  contact_representative_email: "representative@email.com",
  contact_representative_number: "123456789",
  contact_representative_name: "Jill",
  establishment_opening_date: "2018-12-06",
  establishment_type: "Mobile or moveable premise",
  business_type: "Livestock farm",
  business_other_details: "This is the best business in the world",
  opening_day_monday: "Monday"
};

// a supplementary set of all optional answer fields with example data
// (only optional if it's optional within that page. Does not apply to pages that are optional or could be skipped.)
const testOptionalAnswers = {
  establishment_street: "Street name",
  establishment_town: "Town",
  operator_street: "Street name",
  operator_town: "Town",
  contact_representative_role: "Coder",
  operator_charity_number: "123456",
  establishment_secondary_number: "7654321",
  operator_secondary_number: "7654321"
};

// the complete set of possible answer fields with example data
const testComprehensiveAnswers = Object.assign(
  {},
  testMandatoryAnswers,
  testOptionalAnswers
);

// the summary table mounted with the complete set of non-optional answers
const wrapperMinimum = mount(<SummaryTable {...testMandatoryAnswers} />);

// the summary table mounted with the complete set of possible answers
const wrapperComprehensive = mount(
  <SummaryTable {...testComprehensiveAnswers} />
);

const wrapperApplicationComplete = mount(
  <SummaryTable {...testComprehensiveAnswers} applicationCompletePage={true} />
);

describe("<SummaryTable />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<SummaryTable />);
    expect(wrapper.length).toBe(1);
  });

  it("matches the previous snapshot", () => {
    const tree = renderer
      .create(<SummaryTable {...testComprehensiveAnswers} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("gets given all answers as props", () => {
    const props = wrapperComprehensive.props();
    expect(props).toEqual(testComprehensiveAnswers);
  });

  describe("when given a comprehensive set of answers", () => {
    it("the number of table rows matches the allTableRows array", () => {
      const rows = wrapperComprehensive
        .find("Row")
        .findWhere(row => {
          const classNameString = row.prop("className") || "";
          return classNameString.includes("TITLE") === false;
        })
        .find("Row");
      expect(rows.length).toEqual(allTableRows.length);
    });

    describe("when given a props of applicationCompletePage = true", () => {
      it("It doesn't render a change button in all editable rows", () => {
        editableTableRows.forEach(tableRowName => {
          const row = wrapperApplicationComplete.find(`Row#${tableRowName}`);
          const buttonId = `change${tableRowName.charAt(0).toUpperCase() +
            tableRowName.substr(1)}`;
          const button = row.find(`Anchor#${buttonId}`);

          expect(button.length).toBe(0);
        });
      });
    });

    it("renders a change button in all editable rows", () => {
      editableTableRows.forEach(tableRowName => {
        const row = wrapperComprehensive.find(`Row#${tableRowName}`);
        const buttonId = `change${tableRowName.charAt(0).toUpperCase() +
          tableRowName.substr(1)}`;
        const button = row.find(`Anchor#${buttonId}`);

        expect(button.length).toBe(1);
      });
    });

    it("renders all table rows", () => {
      allTableRows.forEach(tableRowName => {
        const row = wrapperComprehensive.find(`Row#${tableRowName}`);
        expect(row.length).toBe(1);
      });
    });

    it("contains a non-empty string for every answer", () => {
      for (let answerID in testComprehensiveAnswers) {
        const text = wrapperComprehensive.find(`#${answerID}`).text();
        expect(text).not.toBe("");
      }
    });
  });

  describe("when given a minimum set of answers", () => {
    it("renders all mandatory table rows with at least one piece of data", () => {
      mandatoryTableRows.forEach(tableRowName => {
        const row = wrapperMinimum.find(`Row#${tableRowName}`);
        expect(row.length).toBe(1);

        const answerCells = row.find("td.summaryTableDataCell");
        expect(answerCells.text()).not.toBe("");
      });
    });

    it("contains empty strings or does not find the element for every optional answer", () => {
      for (let answerID in testOptionalAnswers) {
        const element = wrapperMinimum.find(`#${answerID}`);
        if (element.length !== 0) {
          expect(element.text()).toBe("");
        }
      }
    });
  });
});
