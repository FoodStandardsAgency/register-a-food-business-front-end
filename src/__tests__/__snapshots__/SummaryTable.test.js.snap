// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryTable /> matches the previous snapshot 1`] = `
.emotion-146 {
  border: 0;
  border-collapse: collapse;
  box-sizing: border-box;
  border-spacing: 0;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  line-height: 1.1428571429;
  width: 100%;
}

@media only screen and (min-width:641px) {
  .emotion-146 {
    font-size: 16px;
    line-height: 1.25;
  }
}

.emotion-0 {
  font-family: "nta",Arial,sans-serif;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  text-align: left;
  text-transform: none;
  width: 100%;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    font-size: 19px;
    line-height: 1.3;
  }
}

.emotion-6 {
  display: grid;
  grid-template-columns: 1fr 1fr 70px;
}

.emotion-2 {
  border: 0;
  border-bottom: 1px solid #bfc1c3;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  padding: 0.63158em 1.05263em 0.47368em 0;
  text-align: left;
  text-transform: none;
}

.emotion-2:last-child {
  padding-right: 0;
}

@media only screen and (min-width:641px) {
  .emotion-2 {
    font-size: 19px;
    line-height: 1.3;
  }
}

.emotion-1 {
  font-family: "nta",Arial,sans-serif;
  margin: 0;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 0;
}

.emotion-1 > p {
  margin: 0;
}

.emotion-1 > p > code {
  padding: 0.2em 0.4em;
  margin: 0px;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.emotion-1 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.emotion-1 > pre > code {
  display: inline;
  padding: 0px;
  margin: 0px;
  border: 0px;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    font-size: 19px;
    line-height: 1.3;
  }
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    margin-bottom: 0;
  }
}

.emotion-3 {
  border: 0;
  border-bottom: 1px solid #bfc1c3;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  padding: 0.63158em 1.05263em 0.47368em 0;
  text-align: left;
  text-transform: none;
  word-break: break-word;
}

.emotion-3:last-child {
  padding-right: 0;
}

@media only screen and (min-width:641px) {
  .emotion-3 {
    font-size: 19px;
    line-height: 1.3;
  }
}

.emotion-14 {
  color: #005ea5;
  padding: 3px;
  margin: -3px;
  outline-color: transparent;
  display: inline-block;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-14:focus {
  background-color: #ffbf47;
  outline: 3px solid #ffbf47;
}

.emotion-14:visited {
  color: #2e358b;
}

.emotion-60 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-transform: none;
  font-size: 16px;
  line-height: 1.25;
  color: #6f777b;
}

@media only screen and (min-width:641px) {
  .emotion-60 {
    font-size: 19px;
    line-height: 1.3;
  }
}

<table
  className="emotion-146"
>
  <caption
    className="emotion-0"
  />
  <tbody
    className="emotion-145"
  >
    <tr
      className="TITLE emotion-5 emotion-6"
    >
      <th
        className="emotion-2"
      >
        <div
          className="emotion-1"
        >
          <p>
            <strong>
              Operator details
            </strong>
          </p>
        </div>
      </th>
      <td
        className="emotion-3"
      >
        <div />
      </td>
      <td
        className="emotion-3"
      >
        <div />
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorTypeRow"
    >
      <th
        className="emotion-2"
      >
        Operator type
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="operator_type"
        >
          Sole trader
        </div>
      </td>
      <td
        className="emotion-3"
      >
        <div />
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorCompanyNameRow"
    >
      <th
        className="emotion-2"
      >
        Company name
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="operator_company_name"
        >
          Company name
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/operator-company-details"
          id="changeOperatorCompanyNameRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorCompaniesHouseRow"
    >
      <th
        className="emotion-2"
      >
        Company number
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="operator_company_house_number"
        >
          AA123456
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/operator-company-details"
          id="changeOperatorCompaniesHouseRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorCharityNameRow"
    >
      <th
        className="emotion-2"
      >
        Charity name
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="operator_charity_name"
        >
          Charity name
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/operator-charity-details"
          id="changeOperatorCharityNameRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorCharityNumberRow"
    >
      <th
        className="emotion-2"
      >
        Charity number
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="operator_charity_number"
        >
          123456
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/operator-charity-details"
          id="changeOperatorCharityNumberRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorNameRow"
    >
      <th
        className="emotion-2"
      >
        Name
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
        >
          <span
            id="operator_first_name"
          >
            John
          </span>
           
          <span
            id="operator_last_name"
          >
            Appleseed
          </span>
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/operator-name"
          id="changeOperatorNameRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorAddressRow"
    >
      <th
        className="emotion-2"
      >
        Operator address
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
        >
          <div
            id="operator_first_line"
          >
            Example first line
          </div>
          <div
            id="operator_street"
          >
            Street name
          </div>
          <div
            id="operator_town"
          >
            Town
          </div>
          <div
            id="operator_postcode"
          >
            AA11 1AA
          </div>
        </div>
      </td>
      <td
        className="emotion-3"
      >
        <div />
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorContactDetailsRow"
    >
      <th
        className="emotion-2"
      >
        Phone number
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
        >
          <div
            id="operator_primary_number"
          >
            1234567
          </div>
          <div
            id="operator_secondary_number"
          >
            7654321
          </div>
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/operator-contact-details"
          id="changeOperatorContactDetailsRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="operatorEmailRow"
    >
      <th
        className="emotion-2"
      >
        Email address
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="operator_email"
        >
          operator@email.com
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/operator-contact-details"
          id="changeOperatorEmailRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="contactRepresentativeRow"
    >
      <th
        className="emotion-2"
      >
        Designated contact
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="emotion-60 emotion-61"
        >
          <div
            id="contact_representative_name"
          >
            Jill
          </div>
        </div>
        <div
          className="emotion-60 emotion-61"
        >
          <div
            id="contact_representative_role"
          >
             • Coder
          </div>
        </div>
        <div
          className="bold"
          id="contact_representative_number"
        >
          123456789
        </div>
        <div
          className="bold"
          id="contact_representative_email"
        >
          representative@email.com
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/contact-representative"
          id="changeContactRepresentativeRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="TITLE emotion-5 emotion-6"
    >
      <th
        className="emotion-2"
      >
        <div
          className="emotion-1"
        >
          <p>
            <strong>
              Establishment details
            </strong>
          </p>
        </div>
      </th>
      <td
        className="emotion-3"
      >
        <div />
      </td>
      <td
        className="emotion-3"
      >
        <div />
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="establishmentTradingNameRow"
    >
      <th
        className="emotion-2"
      >
        Trading name
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="establishment_trading_name"
        >
          Example trading name
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/establishment-trading-name"
          id="changeEstablishmentTradingNameRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="establishmentAddressRow"
    >
      <th
        className="emotion-2"
      >
        Establishment address
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
        >
          <div
            id="establishment_first_line"
          >
            Example first line
          </div>
          <div
            id="establishment_street"
          >
            Street name
          </div>
          <div
            id="establishment_town"
          >
            Town
          </div>
          <div
            id="establishment_postcode"
          >
            AA11 1AA
          </div>
        </div>
      </td>
      <td
        className="emotion-3"
      >
        <div />
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="establishmentAddressTypeRow"
    >
      <th
        className="emotion-2"
      >
        Address type
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="establishment_type"
        >
          Mobile or moveable premise
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/establishment-address-type"
          id="changeEstablishmentAddressTypeRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="establishmentContactDetailsRow"
    >
      <th
        className="emotion-2"
      >
        Phone number
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
        >
          <div
            id="establishment_primary_number"
          >
            12345678
          </div>
          <div
            id="establishment_secondary_number"
          >
            7654321
          </div>
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/establishment-contact-details"
          id="changeEstablishmentContactDetailsRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="establishmentEmailRow"
    >
      <th
        className="emotion-2"
      >
        Email address
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="establishment_email"
        >
          establishment@email.com
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/establishment-contact-details"
          id="changeEstablishmentEmailRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="establishmentOpeningDateRow"
    >
      <th
        className="emotion-2"
      >
        Trading date
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="establishment_opening_date"
        >
          06 Dec 2018
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/establishment-opening-status"
          id="changeEstablishmentOpeningDateRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="establishmentOpeningDaysRow"
    >
      <th
        className="emotion-2"
      >
        Opening days
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
        >
          <div
            id="opening_days_irregular"
          >
            <div>
              <div
                id="opening_day_monday"
              >
                Monday
              </div>
              <div
                id="opening_day_tuesday"
              />
              <div
                id="opening_day_wednesday"
              />
              <div
                id="opening_day_thursday"
              />
              <div
                id="opening_day_friday"
              />
              <div
                id="opening_day_saturday"
              />
              <div
                id="opening_day_sunday"
              />
            </div>
          </div>
        </div>
      </td>
      <td
        className="emotion-3"
      >
        <div />
      </td>
    </tr>
    <tr
      className="TITLE emotion-5 emotion-6"
    >
      <th
        className="emotion-2"
      >
        <div
          className="emotion-1"
        >
          <p>
            <strong>
              Activities
            </strong>
          </p>
        </div>
      </th>
      <td
        className="emotion-3"
      >
        <div />
      </td>
      <td
        className="emotion-3"
      >
        <div />
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="activitiesCustomersRow"
    >
      <th
        className="emotion-2"
      >
        Customers
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="customer_type"
        >
          End consumer and Other buisnesses
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/customer-type"
          id="changeActivitiesCustomersRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="businessTypeRow"
    >
      <th
        className="emotion-2"
      >
        Business type
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="business_type"
        >
          Livestock farm
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/business-type"
          id="changeBusinessTypeRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="activitiesBusinessImportExportRow"
    >
      <th
        className="emotion-2"
      >
        Import and export
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="import_export_activities"
        >
          None
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/business-import-export"
          id="changeActivitiesBusinessImportExportRow"
        >
          Change
        </a>
      </td>
    </tr>
    <tr
      className="emotion-5 emotion-6"
      id="businessOtherDetailsRow"
    >
      <th
        className="emotion-2"
      >
        Additional details
      </th>
      <td
        className="summaryTableDataCell emotion-3"
      >
        <div
          className="bold"
          id="business_other_details"
        >
          This is the best business in the world
        </div>
      </td>
      <td
        className="summaryTableChangeCell emotion-3"
      >
        <a
          className="emotion-14"
          href="/edit/business-other-details"
          id="changeBusinessOtherDetailsRow"
        >
          Change
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
